Create a zip packager

derekwbrown opened the pull request:
### Description

Creates a packager (for windows) which simply zips the output directories.

derekwbrown committed:
Add new zip packager

derekwbrown committed:
make extra package dir an array, not a string, so we can zip extra files

KSerrania commented on lib/omnibus/packagers/zip.rb:
```diff
@@ -0,0 +1,163 @@
+
+    
+    #
+    # The path where the MSI resources will live.
```
```suggestion
    # The path where the ZIP resources will live.
```

KSerrania commented on lib/omnibus/packagers/zip.rb:
```diff
@@ -0,0 +1,163 @@
+      raise "Multiple possible matches of `#{search_pattern}'! : #{file_paths}" if file_paths.count > 1
+      file_paths.first.relative_path_from(install_path).to_s
+    end
+    expose :gem_path
```
We don't use `gem_path` here nor in `datadog-agent`. Is it here only to mimic the `.msi` packager or do we really need it?

KSerrania commented on lib/omnibus/packagers/zip.rb:
```diff
@@ -0,0 +1,163 @@
+    #
+    def resources_dir
+      File.expand_path("#{staging_dir}/Resources")
+    end
```
We also don't use the `resources_dir` here. For the `.msi` and `.dmg` packagers, it makes sense as it stores the image and icon files as well as some specification files, but for such a simple packager we can get rid of it (for example, the linux packagers do not have a resources directory).

KSerrania commented on lib/omnibus/packagers/zip.rb:
```diff
@@ -0,0 +1,163 @@
+      #{windows_safe_path(staging_dir)}\\#{project.name}.zip
+      #{windows_safe_path(project.install_dir)}\\*
+      EOH
+    end
```
I think this is also not used. It's identical to the `zip_command` function that is used in the `.msi` packager when used in `fast_msi` mode, so I think it just got copy/pasted, as we have our own zip command above. So I guess either:
- we remove this function, or
- we use it in the build step, with the contents of lines 50-56

KSerrania reviewed (commented):
The linter is complaining in the CI, could you run it to fix the issues?

derekwbrown committed:
Make each individual directory in `extra_package_dir` its own command,
rather than concatenating command lines.
Allows multiple different source directories of the same name (i.e. bin)
to be merged together in a single zip.

Downside of this is that if there's two files of the same name, last one
in wins.

derekwbrown committed:
Review feedback

KSerrania committed:
Linter fixes
